onnxruntime/python/tools/transformers/onnx_exporter.py

```diff
@@ -182,20 +183,40 @@ def optimize_onnx_model(onnx_model_path, optimized_model_path, model_type, num_a
         logger.info(f"Skip optimization since model existed: {optimized_model_path}")
 
 
-def load_pretrained_model(model_name, config, cache_dir):
+def pipeline_dispatcher(model_name, custom_model_class):
+    if (custom_model_class != None):
+        return MODEL_CLASSES[custom_model_class]
+
     if model_name in PRETRAINED_GPT2_MODELS:
-        return GPT2ModelNoPastState.from_pretrained(model_name, config=config, cache_dir=cache_dir)
-    return AutoModel.from_pretrained(model_name, config=config, cache_dir=cache_dir)
+        return GPT2ModelNoPastState
+
+    import re
+    if (re.search('-squad$', model_name) != None):
+        from transformers import AutoModelForQuestionAnswering
+        return AutoModelForQuestionAnswering
+    elif (re.search('-mprc$', model_name) != None):
+        from transformers import AutoModelForSequenceClassification
+        return AutoModelForSequenceClassification
+    elif (re.search('gpt2', model_name) != None):
+        from transformers import AutoModelWithLMHead
+        return AutoModelWithLMHead
+
+    return AutoModel
+
+
+def load_pretrained_model(model_name, config, cache_dir, pipeline):
+    return pipeline.from_pretrained(model_name, config=config, cache_dir=cache_dir)
+
 
+def export_onnx_model(model_name, model_attribute, model_class, cache_dir, onnx_dir, input_names, use_gpu, precision,
+                      optimize_onnx, validate_onnx, use_raw_attention_mask, overwrite, model_fusion_statistics):
+    opset_version, use_external_data_format, model_type = model_attribute[1], model_attribute[2], model_attribute[3]
 
-def export_onnx_model(model_name, opset_version, use_external_data_format, model_type, cache_dir, onnx_dir, input_names,
```

Comment:
It is better to keep old interface that is more straightforward. Consider that this API can be used by others, it is not good to pass model_attribute that might be changed (like add more attribute).